internal/runners/activate/activate.go

```diff
@@ -51,7 +53,14 @@ func (r *Activate) Run(params *ActivateParams) error {
 func (r *Activate) run(params *ActivateParams, activatorLoop activationLoopFunc) error {
 	logging.Debug("Activate %v, %v", params.Namespace, params.PreferredPath)
 
-	pathToUse, err := r.pathToUse(params.Namespace.String(), params.PreferredPath)
+	nSpace := params.Namespace.String()
+	path := params.PreferredPath
+	if params.Replace {
+		nSpace = ""
+		path = ""
+	}
```

Comment:
This isn't necessary. Just drop it. We don't want special behaviour based on some implicit flag.

Reply:
It actually *is* necessary.  If I specify a namespace, `pathToUse` believes that I want to activate a new project and hence need a new path where I can check it out.  So, it runs the `NamespaceSelect.Run()` which asks me for a new project path.

It is ugly either way, because `params.Replace` requires quite a bit of a different behavior:  It makes us re-interpret the namespace argument very differently, and maybe that is wrong in the first place.  As I can see that being very confusing to the user as well.  We could say that `--replace` needs to have a string argument that then will be a completely different parameter `params.ReplaceNamespace`?  Shall I do that? Note, however, that `state activate new-org/new-project --replace` would be invalid then...

Reply:
I see. Good point! I think the cleaner solution is what you proposed: make it an argument. And that also cleans up the UX (makes it less implicit). From a product perspective it shouldn't matter all that much since this is all invoked via the protocol. Which brings up the next problem: if we update this we also have to update the protocol handling. Would you mind doing that for this PR? Shouldn't be much of a detour.